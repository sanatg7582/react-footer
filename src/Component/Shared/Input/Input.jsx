import React from "react";
import PropTypes from "prop-types";
import styles from "./Input.module.scss";
/**
 * function for input type render anywhare
 * @param {string} InputLabel // Name
 * @param {string} InputType // text/number/date
 * @param {string} InputName // name
 * @param {boolean} IsRequired // input required or not
 * @param {fun} handleChange // input value bind function
 * @param {string} InputValue // input value
 * @param {string} InputMaxlength // input maxlength
 * return element with label and Input
 */
const Input = props => {
  const {
    InputLabel,
    InputType,
    InputName,
    IsRequired,
    handleChange,
    InputValue,
    InputMaxlength
  } = props;
  return (
    <div className={styles.InputContainer}>
      {InputLabel ? <label className={styles.labelClass}>{`${InputLabel} :- `}</label> : ""}

      <input
        type={InputType}
        className={styles.formInput}
        name={InputName}
        value={InputValue}
        placeholder={`Please enter ${InputName}`}
        required={IsRequired}
        onChange={e => handleChange(e)}
        maxLength={InputMaxlength}
      />
    </div>
  );
};
// default props
Input.defaultProps = {
  InputLabel: "",
  InputType: "",
  InputName: "",
  IsRequired: false,
  handleChange: () => { },
  InputValue: "",
  InputMaxlength: ""
};
// proptypes validation
Input.propTypes = {
  InputLabel: PropTypes.string.isRequired,
  InputType: PropTypes.string.isRequired,
  InputName: PropTypes.string.isRequired,
  IsRequired: PropTypes.bool,
  handleChange: PropTypes.func,
  InputValue: PropTypes.string,
  InputMaxlength: PropTypes.string
};
export default Input;
