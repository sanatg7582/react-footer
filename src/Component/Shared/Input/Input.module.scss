.InputContainer {
  width: 100%;
  .labelClass {
    width: 100%;
    display: inline-block;
    padding: 5px 0px 5px 0px;
  }
  .formInput {
    width: 100%;
    padding: 5px;
  }
}
