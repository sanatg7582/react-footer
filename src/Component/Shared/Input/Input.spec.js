import React from "react";
import { mount } from "enzyme";
import Input from "./Input";
describe("Load <Input />", () => {
  let wrapper = "";
  beforeAll(() => {
    const props = {
      InputLabel: "ToDo",
      InputType: "text",
      InputName: "todo",
      IsRequired: true,
      handleChange: () => {},
      InputValue: "",
      InputMaxlength: "30"
    };
    wrapper = mount(<Input {...props} />);
    expect(wrapper).toBeTruthy();
  });
  it("renders without crashing", () => {
    expect(wrapper).toBeTruthy();
  });
  it("Input length should be 1", () => {
    expect(wrapper).toHaveLength(1);
  });
  it("checking props", () => {
    expect(
      wrapper
        .find(".InputContainer")
        .find("label")
        .text()
    ).toEqual("ToDo :- ");
    expect(
      wrapper
        .find(".InputContainer")
        .find("input")
        .props().required
    ).toEqual(true);
    expect(
      wrapper
        .find(".InputContainer")
        .find("input")
        .props().type
    ).not.toBe("");
    expect(
      wrapper
        .find(".InputContainer")
        .find("input")
        .type()
    ).toEqual("input");
    expect(
      wrapper
        .find(".InputContainer")
        .find("input")
        .simulate("click")
    );
  });
  it("SnapShot test", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
