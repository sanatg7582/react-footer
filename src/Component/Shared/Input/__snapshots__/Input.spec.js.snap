// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Load <Input /> SnapShot test 1`] = `ReactWrapper {}`;
